tests/auth_rust/src/bin/ckb-auth-cli.rs

```diff
@@ -0,0 +1,159 @@
+            parse_address(blockchain, address);
+            Ok(())
+        }
+        Some(("generate", generate_matches)) => {
+            let address = generate_matches.get_one::<String>("address");
+            let pubkeyhash = generate_matches.get_one::<String>("pubkeyhash");
+            let pubkeyhash = get_pub_key_hash(
+                blockchain,
+                address.as_ref().map(|x| x.as_str()),
+                pubkeyhash.as_ref().map(|x| x.as_str()),
+            )?;
+            generate_message(blockchain, pubkeyhash);
+            Ok(())
+        }
+        Some(("verify", verify_matches)) => {
+            let address = verify_matches.get_one::<String>("address");
+            let pubkeyhash = verify_matches.get_one::<String>("pubkeyhash");
+            let pubkeyhash = get_pub_key_hash(
+                blockchain,
+                address.as_ref().map(|x| x.as_str()),
+                pubkeyhash.as_ref().map(|x| x.as_str()),
+            )?;
+            let signature = verify_matches.get_one::<String>("signature").unwrap();
+            let encoding = verify_matches
+                .get_one::<String>("encoding")
+                .map(String::as_str)
+                .unwrap_or("hex");
+            let signature = decode_string(signature, encoding)?;
+            verify_signature(blockchain, pubkeyhash, signature);
+            Ok(())
+        }
+        _ => {
+            Err(anyhow!("Unknown subcommand"))
+            // Handle invalid or missing subcommands
+        }
+    }
+}
+
+fn decode_string(s: &str, encoding: &str) -> Result<Vec<u8>, Error> {
+    match encoding {
+        "hex" => Ok(hex::decode(s)?),
+        "base64" => {
+            use base64::{engine::general_purpose, Engine as _};
+            Ok(general_purpose::STANDARD.decode(s)?)
+        }
+        _ => Err(anyhow!("Unknown encoding {}", encoding)),
+    }
+}
+fn get_pub_key_hash(
+    blockchain: &str,
+    address: Option<&str>,
+    pubkeyhash: Option<&str>,
+) -> Result<Vec<u8>, Error> {
+    if pubkeyhash.is_some() {
+        return Ok(hex::decode(pubkeyhash.unwrap())?);
+    }
+    if address.is_none() {
+        return Err(anyhow!("Must pass pubkey or pubkeyhash"));
+    }
+    get_pub_key_hash_from_address(blockchain, address.unwrap())
+}
+
+fn get_pub_key_hash_from_address(blockchain: &str, address: &str) -> Result<Vec<u8>, Error> {
+    if blockchain == "litecoin" {
+        // base58 -d <<< mhknqLHQGWDXuLsPdzab8nA4jD3fMdVYS2 | xxd -s 1 -l 20 -p
+        let bytes = bs58::decode(&address).into_vec()?;
+        return Ok(bytes[1..21].into());
+    }
+    Err(anyhow!("Unknown blockchain: {}", blockchain))
+}
+
+fn parse_address(blockchain: &str, address: &str) {
+    println!(
+        "{}",
+        hex::encode(get_pub_key_hash_from_address(blockchain, address).expect("get pub key hash"))
+    );
+}
+
+fn generate_message(_blockchain: &str, pubkeyhash: Vec<u8>) {
+    let algorithm_type = AlgorithmType::Bitcoin;
```

Comment:
"Why is the enumeration type not 'litecoin' even though it was forked from 'bitcoin'?"

Reply:
Thanks for spotting that error. As for use `enum`, this is useful, but not urgent.